common/lib/xmodule/xmodule/js/src/capa/display.js

```diff
@@ -1157,32 +1160,61 @@
          */
         Problem.prototype.disableAllButtonsWhileRunning = function(operationCallback, isFromCheckOperation) {
             var that = this;
-            this.enableAllButtons(false, isFromCheckOperation);
+            var allButtons = [this.resetButton, this.saveButton, this.showButton, this.hintButton, this.submitButton];
+            var disabledButtons = this.disableButtons(allButtons, isFromCheckOperation);
```

Comment:
I think the easiest route that will make the most sense will be to do the filter operation before calling disableButtons, and then not needing the return.